File: src/main/java/com/github/bgora/CarConverter.java
package com.github.bgora;

import com.github.bgora.american.AmericanCar;

public class CarConverter {


    public Car convert(AmericanCar americanCar){
        Integer speed = (int) (americanCar.maxSpeed() * 1.609344);
        return new Car(americanCar.name(),americanCar.horsePower(),speed);
    }
}

File: src/main/kotlin/com/github/bgora/CarConverter.kt
package com.github.bgora

import com.github.bgora.american.AmericanCar

class CarConverter {

    fun convert(americanCar: AmericanCar): Car {
        with(americanCar){
            val converted = maxSpeed * 1.609344
            return Car(name, horsePower, converted.toInt())
        }
    }
}